pandas/core/window.py

```diff
@@ -240,7 +240,7 @@ def _prep_values(self, values: Optional[np.ndarray] = None) -> np.ndarray:
 
         return values
 
-    def _wrap_result(self, result, block=None, obj=None) -> FrameOrSeries:
+    def _wrap_result(self, result, block=None, obj=None):
```

Comment:
Not ideal to remove `FrameOrSeries` here but mypy was complaining since this function is not fully generic (i.e. a conditional branch could determine the returned object)

Options discussed were:
  1. Use Union["Series, "DataFrame"] as a return type
  2. Use NDFrame as a return type
  3. Remove

I looked at option 1 first it caused a lot of flake8 complaints with other imports in the module where we import Series directly within a function, so would add a lot of noqa directives to the file to make it work. Option 2 was suggested by @simonjayhawkins and would be viable but I figured just remove for now and can open up a follow up issue to try and make this function generic. 

It was type checking to `Any` before this change anyway so we aren't loosing any inference. Open to thoughts

Reply:
> It was type checking to `Any` before this change anyway so we aren't loosing any inference.

adding `Any` (or not typing) is not so bad. There is a network effect benefit from typing, so the quicker we add type hints to modules, the quicker we see a benefit.

having `Any` can easily be picked up in a future pass with one of many cli flags to mypy.

so I don't think any PRs adding type annotations should be held pending perfection (and especially unrelated PRs with "Any interest in adding an annotation while you are looking at this?" . xref #26795)

Reply:
Just so we are on the same page anything without an annotation is implicitly `Any` so typing it as such doesn't change anything. 

How do you view the `Any` annotations currently in the code? I interpret that as "this has already been reviewed and really should allow anything" versus "I just added `Any` to get this to pass". I don't see a lot of value add to the latter since that's what happens implicitly without an annotation anyway but certainly a point for discussion

Reply:
> I interpret that as "this has already been reviewed and really should allow anything" versus "I just added Any to get this to pass". I don't see a lot of value add to the latter since that's what happens implicitly without an annotation

+1 for this.  I strongly prefer to leave something un-typed as an invitation to the next person to try to figure it out.

Reply:
Same.

Reply:
> How do you view the `Any` annotations currently in the code?

it depends on the end goal.

I assume that the presence of Any means the module is not yet fully statically typed.

but. this is Python. so why not allow dynamic typing?

if some meaningful work is done in a function/method then the dynamic type will probably still need to conform to some protocol.

Reply:
> +1 for this. I strongly prefer to leave something un-typed as an invitation to the next person to try to figure it out.

just done an experiment.. it appears that Any is considered a type annotation and so the code gets checked, whereas leaving it out causes the type checker to not check the code.

so -1 for not adding Any to un-typed function signatures.

Reply:
What's a case where having a function typed with Any everywhere is valuable?

Reply:
> What's a case where having a function typed with Any everywhere is valuable?

not everywhere!

just where they are currently omitted because the union is of too many types. such as key in indexing.

Reply:
> it appears that Any is considered a type annotation and so the code gets checked, whereas leaving it out causes the type checker to not check the code.

Have to double check but it's not the field you were actually annotated that throws the error as much as something else right? I think mypy might skip a function if it has no annotations so adding Any may cause it to throw errors for inferences of local variables it can't make.

With that said I really don't think adding `Any` is a great approach. If we wanted the former there is a `--check-untyped-def` flag we could enable. 

This thread is probably best as a separate issue

Reply:
> Have to double check but it's not the field you were actually annotated that throws the error as much as something else right? I think mypy might skip a function if it has no annotations so adding Any may cause it to throw errors for inferences of local variables it can't make.

that's correct. or even incorrect calls to typed functions.

> With that said I really don't think adding `Any` is a great approach. 

disagree. 

> If we wanted the former there is a `--check-untyped-def` flag we could enable.

ok locally for a module, but how to set-up on the CI? if we are typing gradually we are not yet ready for this.

Adding Any to a function, indicates the body of that function is ready to be type checked.

> This thread is probably best as a separate issue

i'm opening another PR shortly, where Any is added for all unions of 5 or more items. https://monkeytype.readthedocs.io/en/latest/generation.html#monkeytype.typing.RewriteLargeUnion so we can continue the discusion there.